arkouda/timeclass.py

```diff
@@ -144,6 +139,131 @@ def __init__(self, array, unit: str = _BASE_UNIT):  # type: ignore
         # Deprecated
         self._data = self.values
 
+        if len(self.values):
+            # Year, hour, minute, second
+            days = self.values // _get_factor("d")
+            seconds = self.values // _get_factor("s")
+            days = cast((seconds // 86400), int64) + 1
+            self._year = cast((days // 365.25 + 1970), int64)
```

Comment:
this is being returned as a `pdarray`. In pandas, `year` is returned as an Index. @reuster986, do you have a preference here for pdarray vs index?

Reply:
@Ethan-DeBandi99 good question. I am actually not sure why pandas uses `Index` so much. I don't have a strong opinion either way -- I don't see a compelling need to conform to pandas on this, but if it helps simplify our design philosophy with fewer deviations from pandas, then I'm ok with it. Maybe for now we should stick with `pdarray` and start an issue to explore this question.

Reply:
Sounds good. #1641 added to track the conversation around this.